### PR TITLE
Add tools and docs for changelog generator

### PR DESCRIPTION
**Pull request by Jeffwan**
## Why are these changes needed?

Add tools and docs for changelog generator

## Related issue number

https://github.com/ray-project/kuberay/issues/832

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/release/changelog.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Generate the changelog for a release
```
Comment (DmitriGekhtman) [inline docs/release/changelog.md:1]: Would you mind linking here from the relevant step of `docs/development/release.md`?
(Probably, we should also also move that doc into the `docs/release` into this directory.)

Comment (DmitriGekhtman) [reply docs/release/changelog.md:1]: Never mind, I see you did that in https://github.com/ray-project/kuberay/pull/834/files.
Thanks!

**Review by DmitriGekhtman — APPROVED** [review]
Thanks, this is super helpful!

File: scripts/changelog-generator.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+7f83353 Switch to 0.4.0 and eliminate Chart app versions. (#810)
+86b0af2 Remove ingress.enabled from KubeRay operator chart (#812) (#816)
+c1cbaed Update chart versions for 0.4.0-rc.0 (#804)
+84a70f1 Update image tags. (#784)
+d760b9c [helm] Add memory limits and resource documentation. (#789) (#798)
+16905df [Feature] Improve the observability of integration tests (#775) (#796)
+83aab82 [CI] Pin go version in CRD consistency check (#794) (#797)
+'''
+
+g = ChangelogGenerator("kubeflow/training-operator")
```
Comment (kevin85421) [inline scripts/changelog-generator.py:34]: `ray-project/kuberay`?

File: docs/release/changelog.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+## Prerequisite
+
+1. Prepare your Github Token
+
+1. Install Github python dependencies to generate changlog
```
Comment (kevin85421) [inline docs/release/changelog.md:7]: 2?

File: docs/release/changelog.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+## Generate release notes
+
+1. Run following command and fetch oneline git commits from last release (v0.3.0) to current release (v0.4.0).
+
+    ```
+     git log v0.3.0..v0.4.0 --oneline
+    ```
+ 
+1. Copy above commit history to `scripts/changelog-generator.py` and replace `<your_github_token>` with your Github token. Run the python scripts to generate changelogs.
```
Comment (kevin85421) [inline docs/release/changelog.md:20]: 2?

File: docs/release/changelog.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    83aab82 [CI] Pin go version in CRD consistency check (#794) (#797)
+    ....
+    '''
+
+    g = ChangelogGenerator("ray-project/kuberay")
+    for pr_match in re.finditer(r"#(\d+)", payload):
+        pr_id = int(pr_match.group(1))
+        print("* {}".format(g.generate(pr_id)))
+    ```
+1. You probably need to update the commit logs if it contains tag information or additional PRs links.
```
Comment (kevin85421) [inline docs/release/changelog.md:62]: 3?

File: docs/release/changelog.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+- c1cbaed (tag: v0.4.0-rc.0) Update chart versions for 0.4.0-rc.0 (#804) -> c1cbaed Update chart versions for 0.4.0-rc.0 (#804)
+- 86b0af2 Remove ingress.enabled from KubeRay operator chart (#812) (#816) -> 86b0af2 Remove ingress.enabled from KubeRay operator chart (#816)
+
+1. Cut release from tags and copy results from last step. You can group commits into `Features`, `Bugs` etc and add to release notes.
+See example [v0.3.0 release](https://github.com/ray-project/kuberay/releases/tag/v0.3.0)
+
+1. Send a PR to update [CHANGELOG.md](https://github.com/ray-project/kuberay/blob/master/CHANGELOG.md)
```
Comment (kevin85421) [inline docs/release/changelog.md:70]: 5?

Comment (DmitriGekhtman) [reply docs/release/changelog.md:70]: mkdocs generates correct numbers for the website, but it does look bizarre in the readme

I will address your comments in a follow-up --
(I'm on a plane with nothing to do :) )

Comment (kevin85421) [reply docs/release/changelog.md:70]: Thank @DmitriGekhtman!

**Comment by kevin85421** [comment]
We can also add the link about [creating personal GitHub access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).

**Comment by Jeffwan** [comment]
@kevin85421 @DmitriGekhtman 

Using multiple 1 is a magic to auto rank the number. This is to avoid additional changes for example you delete one line in the middle and you need to rewrite all numbers.

File: scripts/changelog-generator.py (hunk 2 above)
Comment (Jeffwan) [reply scripts/changelog-generator.py:34]: nice cache. I forget to update the repo name. I will file a separate PR to address it

**Comment by DmitriGekhtman** [comment]
> @kevin85421 
> 
> 
> 
> Using multiple 1 is a magic to auto rank the number. This is to avoid additional changes for example you delete one line in the middle and you need to rewrite all numbers. 

Ah, that makes sense. Just looks a little funky when reading the raw file.

**Comment by DmitriGekhtman** [comment]
> @kevin85421 
> 
> 
> 
> Using multiple 1 is a magic to auto rank the number. This is to avoid additional changes for example you delete one line in the middle and you need to rewrite all numbers. 

Ah, that makes sense. Just looks a little funky when reading the raw file.

**Pull request merged**